common4j/src/main/com/microsoft/identity/common/java/platform/AbstractDevicePopManager.java

```diff
@@ -986,6 +993,22 @@ private String mintSignedHttpRequestInternal(@Nullable final String httpMethod,
             exception = e;
             errCode = KEYSTORE_NOT_INITIALIZED;
         } catch (final JOSEException e) {
+            if ((isNegativeInternalError(e)) ||
+                    (e.getCause() != null && isNegativeInternalError(e.getCause())) ||
+                    (e.getCause().getCause() != null && isNegativeInternalError(e.getCause().getCause()))
```

Comment:
getCause().getCause()? Is this required?

Reply:
yes this is for the 3rd level of the exception

this is the stack trace
Caused by: com.nimbusds.jose.JOSEException: Invalid private RSA key: Keystore operation failed
		at com.nimbusds.jose.crypto.RSASSASigner.getInitiatedSignature(:263)
		at com.nimbusds.jose.crypto.RSASSASigner.sign(:236)
		at com.nimbusds.jose.JWSObject.sign(:304)
		at com.microsoft.identity.common.java.platform.AbstractDevicePopManager.mintSignedHttpRequestInternal(:979)
		... 16 more
	Caused by: java.security.InvalidKeyException: Keystore operation failed
		at android.security.keystore2.KeyStoreCryptoOperationUtils.getInvalidKeyException(KeyStoreCryptoOperationUtils.java:130)
		at android.security.keystore2.AndroidKeyStoreSignatureSpiBase.ensureKeystoreOperationInitialized(AndroidKeyStoreSignatureSpiBase.java:217)
		at android.security.keystore2.AndroidKeyStoreSignatureSpiBase.engineInitSign(AndroidKeyStoreSignatureSpiBase.java:123)
		at android.security.keystore2.AndroidKeyStoreSignatureSpiBase.engineInitSign(AndroidKeyStoreSignatureSpiBase.java:101)
		at java.security.Signature$Delegate.init(Signature.java:1357)
		at java.security.Signature$Delegate.chooseProvider(Signature.java:1310)
		at java.security.Signature$Delegate.engineInitSign(Signature.java:1385)
		at java.security.Signature.initSign(Signature.java:679)
		at com.nimbusds.jose.crypto.RSASSASigner.getInitiatedSignature(:261)
		... 19 more
	Caused by: android.security.KeyStoreException: Invalid key blob (internal Keystore code: -33 message: In create_operation: Failed to begin operation. 1010266